File: 1-easy/problems/0868-binary-gap/solution.ts
function binaryGap(N: any): boolean | number | string | any {
  """
  :type N: int
  :rtype: int
  """
  result = 0
  last = null
  for (i in Array.from({length: 32}, (_, i) => i)) {
      if ((N >> i) & 1) {
          if (last is not null) {
              result = max(result, i-last)
          last = i
  return result;
}

export { binaryGap };

File: 1-easy/problems/0868-binary-gap/solution.py
class Solution:
    def binaryGap(self, N):
        """
        :type N: int
        :rtype: int
        """
        result = 0
        last = None
        for i in range(32):
            if (N >> i) & 1:
                if last is not None:
                    result = max(result, i-last)
                last = i
        return result
